### ISSUE TITLE
Bug causing sprite to be oriented backwards (flipped 180 around Y axis)

### ISSUE THREAD

**Issue by jwzimmer** [LLK/scratch-flash#1305]
**TL;DR:** Even though the two blue worms are oriented in the same direction programmatically and in the thumbnails, in the project one of them shows up flipped around the Y axis.

If you set both sprites to point in the direction 90 degrees, they still point in opposite directions. I'm not sure how the project got into this state, but it seems like triggering a save (from the paint editor?) (?) might correct the bug.

@christanbalch suggested I talk to @ntlrsk since it looks like a rotation-related bug. We looked at the project and she said:
> It’s something about the vector editor - if you delete the red bar (or another part of the sprite) it corrects. Even if you add a line (or move the points around) it corrects. And if you switch to bitmap it corrects.

I wondered if maybe when they created the backwards sprite in the vector editor, they dragged the points around until the shape was inverted from it's original rotation (as in, when they first dragged one of the component shapes out, point A was on the left of the screen and point B was on the right, and when they finished arranging the points into the shape they wanted, point A was on the right and point B was on the left, something like that)?

To repro, press the green flag in 157943220 and check out the sprite orientations for the blue worm sprites. You can use the [point in direction (90)] block to see that they point in opposite directions.

When I remix my remix of the project as 159699366 or 159699354, the sprites are **no longer oriented backwards**...? So be careful when using the project, since I don't want to lose the ability to repro the bug. : )

<img width="567" alt="screen shot 2017-05-08 at 3 46 07 pm" src="https://cloud.githubusercontent.com/assets/1320456/25822145/abe25ffe-3405-11e7-89e9-7cea0388f724.png">

**Comment by TheLogFather** [comment]
Sounds like the Stage3D flipped sprite bug to me.

See demo project & explanation of how it happens:  https://scratch.mit.edu/projects/152407373/

You should be able to work around it by ensuring the sprites are oriented facing right (i.e. 90º) whenever the project is saved.

**Comment by TheLogFather** [comment]
Huh, for some reason I can no longer edit my comments on github... :/

Just wanted to add that I couldn't find a previous mention of it in the github issues (which surprised me, since it's been known about and mentioned in the forums for quite some time), so I guess this is now the 'official' bug report for it...  :)

**Comment by TheLogFather** [comment]
Ah, I found an earlier bug report for this:  https://github.com/LLK/scratch-flash/issues/708
Didn't get very far, though, 'cos it's not obvious how to repro (hopefully, my demo project helps there).

**Comment by thisandagain** [comment]
Tracking in https://github.com/LLK/scratch-flash/issues/708
